rfcs/0052-create-sink-into-table.md

```diff
@@ -0,0 +1,400 @@
+But RW can not give consistent guarantee, so user must give additional declaration in SQL.
+
+I summarize two patterns here
+- Backfill: If user have all historical data in the upstream table, he can drop the old the create the new sink to the table
+  - 👍 Easy to understand. No more SQL rewritten
+  - 👍 eventually consistency
+  - 👎 The result table and all its down stream mv's version will rollback and seems a stale read for user.
+  - 👎 Long time to backfill and re-compute all the data.
+  - 👎 In some cases, user can not get all upstream historical data such as a kafka source with TTL.  
+- Apply delta: User can determine a timestamp of event time to determine use the old sink or the new sink to update the result table.
+  - 👍 No result rollback
+  - 👍 Incremental computation
+  - 👎 Limited with event time column in the result table.
+  - 👎 Might be complex for user to write on conflict.
+
+Examples :
+1. This is a Backfill case but user define a proper on conflict behavior with the monotonically increasing `order_cnt` to prevent data rollback. 
+```SQL
+  CREATE TABLE dwd_orders (
+    user VARCHAR,
+    product VARCHAR,
+    price BIGINT,
+    event_time TIMESTAMP,
+) APPEND ONLY;
+
+CREATE TABLE dws_product (
+  product VARCHAR,
+  order_cnt BIGINT,
+  PRIMARY KEY (product)
+) WITH (
+  'ttl' = '7d'
+  -- only update when the new incoming row's order_cnt is larger than the old order_cnt
+) ON CONFLICT DO UPDATE SET (product, order_cnt) = (
+      CASE WHEN EXCLUDED.order_cnt IS NOT NULL AND EXCLUDED.order_cnt > order_cnt 
+        THEN EXCLUDED.product ELSE product 
+      END,
+      CASE WHEN EXCLUDED.order_cnt IS NOT NULL AND EXCLUDED.order_cnt > order_cnt 
+        THEN EXCLUDED.order_cnt, order_cnt ELSE order_cnt 
+      END)
+
+CREATE SINK order_production_analyze_job INTO dws_product AS 
+SELECT 
+  window_start,
+  product,
+  COUNT(*) as order_cnt,
+FROM dwd_orders
+GROUP BY product
+WITH (
+  -- because the table's on- conflict behavior can only applied on the “INSERT” conflict，so we need use force append-only sink here.
+  force_append_only = 'true'
+);
+
+-- when the user want to add an new aggregator
+ALTER TABLE dws_product ADD sell_sum BIGINT ON CONFLICT DO UPDATE SET 
+  sell_cnt = CASE WHEN EXCLUDED.order_cnt IS NOT NULL AND EXCLUDED.order_cnt > order_cnt 
+                  THEN EXCLUDED.sell_sum ELSE sell_sum 
+             END;
+CREATE SINK new_order_production_analyze_job INTO dws_product AS 
+SELECT 
+  window_start,
+  product,
+  COUNT(*) as order_cnt,
+  sum(price) as sell_sum,
+FROM dwd_orders
+GROUP BY window_start, product
+WITH (
+  force_append_only = 'true'
+);
+
+-- when the new job catch up
+DROP SINK order_production_analyze_job;
+ALTER SINK new_order_production_analyze_job RENAME TO order_production_analyze_job;
+```
+
+2. Apply delta
```

Comment:
This also looks too complicated to me. All these 100 lines of SQL achieve the exactly same stuff as `CREATE OR REPLACE MATERIALIZED VIEW AS <add new agg call>`, where I don't think there's any need to expose such details to users.

Reply:
On the other side, the new sink doesn't have access to the states in the old sink, so this can be hardly applied to stateful queries.

Reply:
> On the other side, the new sink doesn't have access to the states in the old sink, so this can be hardly applied to stateful queries.

I guess this restriction is requiring a event time column in the result table so that user can divide their result set :thinking: